milabench/common.py

```diff
@@ -296,6 +308,160 @@ def _read_reports(*runs):
+    if local_meta:
+        local_meta = local_meta["data"]
+    remote_metas = []
+    for _r in reports:
+        meta_lines = []
+        for event in _r:
+            _, event_type, line = None, "", []
+
+            try:
+                _, event_type, *line = event["data"].split(" ")
+            except (AttributeError, ValueError):
+                pass
+
+            if event_type[:1] + event_type[-1:] != "[]":
+                event_type = None
+                line = event["data"]
+            else:
+                line = " ".join(line)
+
+            if event_type == "[meta]":
+                meta_lines.append(line)
+            elif event_type is None and meta_lines:
+                meta_lines.append(line)
+            elif meta_lines:
+                remote_metas.append(yaml.safe_load("".join(meta_lines)))
+                meta_lines = []
+
+    return local_meta, remote_metas
+
+
+def _filter_reports(*reports):
+    all_reports = []
+
+    for report in reports:
+        config = next(iter(e for e in report if e["event"] == "config"), None)
+        if config is None:
+            continue
+
+        if config["data"]["name"] != "remote":
+            all_reports.append(report)
+
+    return all_reports
+
+
+def _push_reports(reports_repo, runs, packs:dict=None):
+    _SVG_COLORS = {
+        "pass": "blue",
+        "partial": "yellow",
+        "failure": "red",
+    }
+    import milabench.cli.badges as badges
+
+    _repo = git.repo.base.Repo(ROOT_FOLDER)
+    try:
+        reports_repo = git.repo.base.Repo(str(reports_repo))
+    except (git.exc.InvalidGitRepositoryError, git.exc.NoSuchPathError):
+        repo_url = next(iter(_r.url for _r in _repo.remotes if _r.name == "origin"), None)
+        reports_repo = git.repo.base.Repo.clone_from(repo_url, str(reports_repo), branch="reports")
+
+    reports_url = ([
+        _r.url for _r in _repo.remotes if "mila-iqia" in _r.url
+    ] or [
+        _r.url for _r in _repo.remotes if _r.name == "origin"
+    ])[0]
+    reports_url = XPath("github.com".join(reports_url.split("github.com")[1:])[1:])
+    reports_url = XPath("https://github.com") / f"{reports_url.with_suffix('')}/tree/{reports_repo.active_branch.name}"
+
+    device_reports = {}
+    for run in runs:
+        reports = list(_read_reports(run).values())
+        reports = _filter_reports(*reports)
+
+        if not reports:
+            continue
+
+        meta = [e["data"] for _r in reports for e in _r if e["event"] == "meta"]
+
+        for _meta in meta:
+            for gpu in _meta["accelerators"]["gpus"].values():
+                device = gpu["product"].replace(" ", "_")
+                break
+        else:
+            for _meta in meta:
+                device = _meta["cpu"]["brand"].replace(" ", "_")
+                break
+
+        tag = ([
+            t.name
+            for t in _repo.tags
+            if meta[0]["milabench"]["tag"].startswith(t.name)
+        ] or [meta[0]["milabench"]["tag"]])[0]
```

Comment:
I kind of feel that using the tag as is would be better even if there's a chance I think it will duplicate the versions? Let me know

```suggestion
        tag = meta[0]["milabench"]["tag"]
```

Reply:
There is [this](https://github.com/mila-iqia/milabench/blob/master/milabench/scripts/vcs.py#L21)


I think it would be clearer if we use  "branch-commit"  because the tag means actually nothing in most cases.
For example, in master the latest tag is still v0.0.6 but stable latest tag is v0.0.10.